src/Conference/ConferenceDetails.jsx

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import styled from "styled-components";
+import { getTimestring, getDatestring } from "../utils";
+import { Link, useParams } from "react-router-dom";
+
+const StyledConferenceDetails = styled.div`
+  width: 100%;
+  height: 400px;
+  color: white;
+  background: darkslategray;
+`;
+
+const ConferenceDetails = ({ conferences }) => {
+  let { id } = useParams();
+  let conference = {};
+  conferences.forEach(singleConference => {
```

Comment:
Javascript arrays have a method called find that given a function returns the object(s) that pass this value.

So in your case you'd write:

const conference = conferences.find(conference => conference.id ===id)

Reply:
Agreed, find would be nicer here :) Although you'll still need to convert the id to a number, using parseInt or Number as you already are.

Reply:
Hi Dex, just a bit confused as to where the "const conference = conferences.find(conference => conference.id ===id)" should be used. We were assuming it was in place of the if statement?